src/requests/cache/sorted_set/sorted_set_put_elements.rs

```diff
@@ -1,17 +1,104 @@
+use std::collections::HashMap;
+
 use momento_protos::cache_client::{SortedSetElement, SortedSetPutRequest};
 
 use crate::requests::cache::MomentoRequest;
 use crate::simple_cache_client::prep_request_with_timeout;
 use crate::{CacheClient, CollectionTtl, IntoBytes, MomentoResult};
 
+/// This trait defines an interface for converting a type into a vector of `SortedSetElement`s.
+pub trait IntoSortedSetElements: Send {
+    /// Converts the type into a vector of `SortedSetElement`s.
+    fn into_sorted_set_elements(self) -> Vec<SortedSetElement>;
+}
+
+/// Converts an iterator of value-score pairs into a vector of `SortedSetElement`s.
+///
+/// # Arguments
+///
+/// - `iter`: An iterator over pairs `(V, f64)` where `V` is a value that implements
+///   the `IntoBytes` trait, and `f64` represents the score associated with the value.
+///
+/// # Returns
+///
+/// A `Vec<SortedSetElement>` where each `SortedSetElement` contains a byte representation
+/// of the value and its associated score. The order of elements in the returned vector
+/// matches the order of pairs in the input iterator.
+///
+/// # Examples
+///
+/// Basic usage with a vector of tuples:
+///
+/// ```
+/// use momento::requests::cache::sorted_set::sorted_set_put_elements::map_and_collect_sorted_set_elements;
+/// let pairs = vec![("value1", 1.0), ("value2", 2.0)];
+/// let sorted_set_elements = map_and_collect_sorted_set_elements(pairs.into_iter());
```

Comment:
Can they not just call `pairs.into()` here?

Reply:
Yeah but this is documentation of the utility method not the trait. I put this here in detail for anyone else who may wish to implement the trait for their own custom container.

Reply:
based on my comment above about my attempt to do `From`, I think the compiler won't let people impl this trait for other collection types. which seems like a bummer... I hope I'm missing something.

Reply:
I also confirmed not possible due to the reasons you mentioned below.